background.js

```diff
@@ -1,124 +1,101 @@
 (function () {
-    const URL_REGEX = /^https?:\/\/docs\.python\.org\/2[^\/]*?\/(.*)/;
-    const URL_REPLACEMENT = "https://docs.python.org/3/$1";
-
+    const URL_REGEX = /^(https?:\/\/docs\.python\.org\/)([23][^\/]*?)(\/.*)/;
     const SPECIAL_CASES = {
-        'library/sets.html' : 'library/stdtypes.html#set',
-        'library/stringio.html': 'library/io.html#io.StringIO'
+        "/library/sets.html" : "/library/stdtypes.html#set",
+        "/library/stringio.html": "/library/io.html#io.StringIO"
     };
 
-    let isEnabled = true;
-    updateIsEnabled();
+    var pyVersion, isEnabled;
+
+    browserAPI.api.storage.local.get(
+        {pyVersion: "3", isEnabled: true},
+        Object.assign.bind(undefined, this)
```

Comment:
Note that this doesn't work as you expect. `this` here is `window`, not the local scope of the anonymous function. What happens here is you basically ignore `pyVersion` and `isEnabled` defined earlier, and create new fields in `window`, effectively rendering the variables defined above useless and not used at all. While this works on Chrome, it does not on Firefox - it uses the variables above (which are always `undefined`) instead of those in `window`, which mean that the Firefox version ignores whatever user sets (and therefore never redirects).